src/MatrixManifold.jl

```diff
@@ -0,0 +1,29 @@
+"""
+    MatrixManifold{M <: Manifold} <: Manifold
+
+A manifold to encapsulate manifolds working on array representations of
+`MPoints` and `TVectors` in a transparent way, such that for these manifolds its
+not necessary to introduce explicit types for the points and tangent vectors,
+but they are encapusalted/stripped automatically when needed.
+"""
+struct MatrixManifold{M <: Manifold}
```

Comment:
It's a bit confusing calling it a manifold when it doesn't inherit `Manifold`.

Reply:
Oh sorry, forgot that :(